src/Data/Bool/Compat.hs

```diff
@@ -12,6 +15,41 @@ module Data.Bool.Compat (
 import Data.Bool
 
 #if !MIN_VERSION_base(4,7,0)
+
+-- These instances are only valid if Bits isn't a subclass of Num (as Bool is
+-- not a Num instance), which is only true as of base-4.6.0.0 and later.
+# if MIN_VERSION_base(4,6,0)
+import Data.Bits (Bits(..))
+import Data.Eq ((/=))
+
+instance Bits Bool where
```

Comment:
What about moving this to `Data.Bits.Compat`?  My reasoning is:

 * In `base` the instance is defined in `Data.Bits`
 * To use the instance a user has to import `Data.Bits`/`Data.Bits.Compat` anyway.  Moving the instance to `Data.Bits.Compat` allows the user to use the instance with only one import instead of two.